test/unit/constraints/boolean.test.ts

```diff
@@ -22,16 +22,19 @@ describe("Boolean constraints", function () {
             expect(metadata).toEqual(expected);
         });
 
-        it("should error when applied to a non-boolean property", function () {
-            expect(function () {
-                class MyBadClass {
-                    @BooleanSchema()
-                    myBadProperty! : number;
-
-                    @BooleanSchema()
-                    myOtherBadProperty! : string;
-                }
-            }).toThrow(ConstraintDefinitionError);
+        /**
+         * TODO: test compilation failures
```

Comment:
I'm curious how you'd approach testing compilation failures. Have a test helper that accepts source code as a string and runs the compiler API / `tsc` and checking the errors. I'd be interested in the implementation as I have a similar need in other libs.

Reply:
Yes, I think that's the only way that'll work. I believe there are libs out that to do it.